crates/db_schema/src/impls/password_reset_request.rs

```diff
@@ -52,23 +32,27 @@ impl PasswordResetRequest {
       local_user_id: from_local_user_id,
       token: token_,
     };
-
-    Self::create(pool, &form).await
+    let conn = &mut get_conn(pool).await?;
+    insert_into(password_reset_request)
+      .values(form)
+      .get_result::<Self>(conn)
+      .await
   }
-  pub async fn read_from_token(pool: &mut DbPool<'_>, token_: &str) -> Result<Option<Self>, Error> {
+
+  /// Reads reset token and invalidates it
+  pub async fn read(pool: &mut DbPool<'_>, token_: &str) -> Result<Option<Self>, Error> {
     let conn = &mut get_conn(pool).await?;
-    password_reset_request
+    update(password_reset_request)
```

Comment:
Use delete instead of update

Reply:
We need to keep the column so that reset count works. That one is used to limit users to 3 password resets per day. However we could get rid of that and use a normal rate limit instead, eg `auth` rate limit which applies to login, register and password reset.

Reply:
Per-user rate limits are better